### PR TITLE
Introduce `SimplifiedKotlinScriptEvaluator` for cheaper DSL testing

### PR DESCRIPTION
**Pull request by bamboo**

### CODE REVIEW

File: subprojects/provider/src/test/kotlin/org/gradle/kotlin/dsl/execution/SimplifiedKotlinScriptEvaluator.kt [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+    SimplifiedKotlinScriptEvaluator(baseCacheDir).use {
+        it.eval(script, target)
+    }
+}
+
+
+/**
+ * A simplified Kotlin script evaluator, suitable for cheaper testing of the DSL outside Gradle.
+ */
+class SimplifiedKotlinScriptEvaluator(
```
Comment (eskatos) [inline subprojects/provider/src/test/kotlin/org/gradle/kotlin/dsl/execution/SimplifiedKotlinScriptEvaluator.kt]: ❓ could we move that guy into `:test-fixtures`?

File: subprojects/provider/src/test/kotlin/org/gradle/kotlin/dsl/execution/SimplifiedKotlinScriptEvaluatorTest.kt [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+        val project = mock<Project>()
+        eval(
+            script = """
+                version = "1.0"
+            """,
+            target = project
+        )
+        verify(project).version = "1.0"
+    }
```
Comment (eskatos) [inline subprojects/provider/src/test/kotlin/org/gradle/kotlin/dsl/execution/SimplifiedKotlinScriptEvaluatorTest.kt]: Lovely 👍

**Comment by bamboo** [comment]
@eskatos done moving `SimplifiedKotlinScriptEvaluator` to `:test-fixtures`. Unfortunately this required `Interpreter` and `ProgramId` to be made public (a fair price to pay). Please take another look.

I'll rename `:test-fixtures` to `:testing` and reorganize the packages once this is merged.

File: subprojects/provider/src/main/kotlin/org/gradle/kotlin/dsl/execution/Interpreter.kt [hunk 3]
```diff
@@ -67,7 +67,6 @@ import java.lang.reflect.InvocationTargetException
  * @see ResidualProgram
  * @see ResidualProgramCompiler
  */
-internal
```
Comment (eskatos) [inline subprojects/provider/src/main/kotlin/org/gradle/kotlin/dsl/execution/Interpreter.kt]: ❓ any chance there are now public members that could be made `internal` to limit the surface?

Comment (bamboo) [reply subprojects/provider/src/main/kotlin/org/gradle/kotlin/dsl/execution/Interpreter.kt]: `ProgramId` exposes no members and `Interpreter` exposes only `eval` and the `Host` interface, both required by the testing infrastructure.

**Pull request merged**